lisa/tools/kdump.py

```diff
@@ -0,0 +1,432 @@
+    is not performed during a kexec boot. This has the effect of reducing the
+    time required for a reboot.
+    """
+
+    # kexec-tools 2.0.16
+    __pattern_kexec_version_info = re.compile(
+        r"^kexec\S+\s+(?P<major>\d+).(?P<minor>\d+).(?P<patch>\d+)"
+    )
+
+    # Existed bug for kexec-tools 2.0.14
+    # https://bugs.launchpad.net/ubuntu/+source/kexec-tools/+bug/1713940
+    # If the version of kexec-tools is lower than 2.0.15, we install kexec from source
+    _target_kexec_version = "2.0.15"
+
+    # If install kexec from source, we choose 2.0.18 version for it is stable for most
+    # Debian distros
+    _kexec_repo = (
+        "https://mirrors.edge.kernel.org/pub/linux/utils/kernel/kexec/"
+        "kexec-tools-2.0.18.tar.gz"
+    )
+
+    @property
+    def command(self) -> str:
+        return "kexec"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        assert isinstance(self.node.os, Posix)
+        self.node.os.install_packages("kexec-tools")
+        if isinstance(self.node.os, Debian):
+            version = self._get_version()
+            if version < self._target_kexec_version:
+                self._install_from_src()
+        return self._check_exists()
+
+    def _get_version(self) -> VersionInfo:
+        result = self.run(
+            "-v",
+            force_run=False,
+            no_error_log=True,
+            no_info_log=True,
+            sudo=True,
+            shell=False,
+        )
+        result.assert_exit_code(message=result.stderr)
+        raw_version = re.finditer(self.__pattern_kexec_version_info, result.stdout)
+        for version in raw_version:
+            matched_version = self.__pattern_kexec_version_info.match(version.group())
+            if matched_version:
+                major = matched_version.group("major")
+                minor = matched_version.group("minor")
+                patch = matched_version.group("patch")
+                self._log.info(f"kexec version is {major}.{minor}.{patch}")
+                return VersionInfo(int(major), int(minor), int(patch))
+        raise LisaException("No find matched kexec version")
+
+    def _install_from_src(self) -> None:
+        tool_path = self.get_tool_path()
+        self.node.shell.mkdir(tool_path, exist_ok=True)
+        wget = self.node.tools[Wget]
+        kexec_tar = wget.get(self._kexec_repo, str(tool_path))
+        tar = self.node.tools[Tar]
+        kexec_source = tar.extract(f"{kexec_tar}", str(tool_path))
+        code_path = tool_path.joinpath(kexec_source)
+        self.node.tools[Gcc]
+        make = self.node.tools[Make]
+        self.node.execute("./configure", cwd=code_path).assert_exit_code()
+        make.make_install(cwd=code_path)
+        self.node.execute(
+            "yes | cp -f /usr/local/sbin/kexec /sbin/", sudo=True, shell=True
+        ).assert_exit_code()
+
+
+class Makedumpfile(Tool):
```

Comment:
Where is it used, rather than referenced as a dependency in Kdump?

Reply:
It is only referenced as a dependency in Kdump.

Reply:
The same as kexec